### PR TITLE
feat: replace Engine by dagger.Connect()

### LINKED ISSUE #3607: NodeJS SDK - Replace Engine by dagger.connect
As mentionned by @aluzzardi here https://github.com/dagger/dagger/pull/3460#issuecomment-1292758263. 
NodeJS SDK need to user `dagger.connect` instead of `Engine()`.

### PR DESCRIPTION
**Pull request by TomChv**
**Description**

Implement a new function to initialize Dagger
Cloak client.
This implementation is based on the old
Typescript Engine but with code improvements.

**TODO**
- [x] Find a solution to cancel the server (shall we implement a close method on the `Client`)?
- [x] Add tests
- [ ] Add Unit tests on subfunctions

Resolves #3607

### CODE REVIEW

File: sdk/nodejs/dagger/connect.ts
Comment (TomChv) [inline sdk/nodejs/dagger/connect.ts:56]: @slumbering I need your extra attention on this part, I'm waiting for your opinion before implementing this part

File: sdk/nodejs/dagger/connect.ts
Comment (TomChv) [inline sdk/nodejs/dagger/connect.ts:18]: According to https://github.com/dagger/dagger/pull/3265#discussion_r1010930750, shall we remove it in Ts SDK too?

Comment (slumbering) [reply sdk/nodejs/dagger/connect.ts:18]: I would rather wait for the PR #3265 to be ready and merged. Then we can get rid of this part.

File: sdk/nodejs/dagger/connect.ts
Comment (slumbering) [inline sdk/nodejs/dagger/connect.ts:35]: nit: Any reason of using the underscore in front of config ? Why not using another name like defaultConfig ? AFAIK the underscore is a convention to declare private variables.

File: sdk/nodejs/dagger/connect.ts
Comment (slumbering) [inline sdk/nodejs/dagger/connect.ts:138]: nit: having a const here is not necessary IMO. Using the `500` in its place would be more straightforward.

Comment (TomChv) [reply sdk/nodejs/dagger/connect.ts]: Updated in https://github.com/dagger/dagger/pull/3652/commits/8c2c0da71b28e8c37547b10b65cb5fe577583432

File: sdk/nodejs/dagger/client.ts
Comment (TomChv) [inline sdk/nodejs/dagger/client.ts:17]: I could not find why but the client does not work when it's using socket...
That only work when I use `GraphQLClient` that hits `http://localhost${port}`.

File: sdk/nodejs/test/index.mjs
Comment (slumbering) [inline sdk/nodejs/test/index.mjs:36]: So this mean we need to call manually the close function to kill the child process right ? 🤔 
If it's the case I would rather call close() implicitly after a Graphql request.

Comment (TomChv) [reply sdk/nodejs/test/index.mjs:36]: Since there's no callback yeah, I could also add a special argument to the "do" function so you can close the client after the query.
Or you could bind a .finally after called do to close the client.

Comment (slumbering) [reply sdk/nodejs/test/index.mjs:36]: @TomChv has more context now and work on another way to handle child process killing

File: sdk/nodejs/dagger/connect.ts
Comment (slumbering) [reply sdk/nodejs/dagger/connect.ts:18]: Actually you're right, let's get rid of it.

**Review by slumbering — CHANGES_REQUESTED** [review]

File: sdk/nodejs/dagger/connect.ts
Comment (TomChv) [reply sdk/nodejs/dagger/connect.ts]: Done in https://github.com/dagger/dagger/pull/3652/commits/5d6905c876441a1cae0453ae5402b37aeeb83c24

File: sdk/nodejs/test/index.mjs
Comment (TomChv) [reply sdk/nodejs/test/index.mjs]: Done in https://github.com/dagger/dagger/pull/3652/commits/5d6905c876441a1cae0453ae5402b37aeeb83c24

File: sdk/nodejs/dagger/client.ts
Comment (slumbering) [reply sdk/nodejs/dagger/client.ts:17]: Can you create an issue for that please @TomChv  🙏 ? Let's move forward on this PR

**Review by slumbering — APPROVED** [review]

**Comment by slumbering** [comment]
LGTM! Thanks @TomChv

**Comment by aluzzardi** [comment]
Please double check before merging PRs -- there's a bad commit that got merged into main:

`Merge branch 'main' into feat/ts-sdk-replace-engine-by-dagger-connect`

**Comment by slumbering** [comment]
> Please double check before merging PRs -- there's a bad commit that got merged into main:
> 
> `Merge branch 'main' into feat/ts-sdk-replace-engine-by-dagger-connect`

This commit is created when you click on `Update branch`. 

Capture d’écran 2022-11-08 à 09 06 30

Should I choose `Update With Rebase` next time ? Sorry for the bad commit

**Pull request merged**
